gtsfm/two_view_estimator_cacher.py

```diff
@@ -50,6 +50,8 @@ def __generate_cache_key(
         self, keypoints_i1: Keypoints, keypoints_i2: Keypoints, putative_corr_idxs: np.ndarray
     ) -> str:
         """Generates a cache key according to keypoint coordinates and putative correspondence indices."""
+        if putative_corr_idxs.size == 0:  # catch no correspondences
+            return cache_utils.generate_hash_for_numpy_array(np.array([]))
```

Comment:
could we add a brief comment explaining why our existing logic won't handle this case? Could we add a test also?